examples/zkapps/06-offchain-storage/offchain-storage-zkapp/ui/pages/index.page.tsx

```diff
@@ -0,0 +1,303 @@
+
+  const onRefreshCurrentNum = async () => {
+    console.log('getting zkApp state...');
+    await state.zkappWorkerClient!.fetchAccount({
+      publicKey: state.zkappPublicKey!,
+    });
+    const currentMessages = await state.zkappWorkerClient!.getMessages();
+    console.log('current state:', currentMessages.toString());
+
+    setState({ ...state, currentMessages });
+  };
+
+  // -------------------------------------------------------
+  // Create UI elements
+
+  let setupText = state.hasBeenSetup
+    ? 'SnarkyJS Ready'
+    : 'Setting up SnarkyJS...';
+  let setup = <div> {setupText} </div>;
+
+  let accountDoesNotExist;
+  if (state.hasBeenSetup && !state.accountExists) {
+    const faucetLink =
+      'https://faucet.minaprotocol.com/?address=' +
+      state.feePayerPublicKey!.toBase58();
+    accountDoesNotExist = (
+      <div>
+        Account does not exist. Please visit the faucet to fund this account
+        <a href={faucetLink} target="_blank" rel="noreferrer">
+          {' '}
+          [Link]{' '}
+        </a>
+      </div>
+    );
+  }
+
+  let mainContent;
+  if (state.hasBeenSetup && state.accountExists) {
+    let messages = [];
+    for (let i = 0; i < 2 ** (8 - 1); i++) {
+      let input;
+      let onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const userMessages = [...state.userMessages];
+        userMessages[i] = event.target.value;
+        setState({ ...state, userMessages });
+      };
+
+      let signer;
+      if (state.currentMessages.has(i)) {
+        const value = state.currentMessages.get(i)!;
+        const publicKeyFields = value.slice(0, 2);
+
+        const publicKey = PublicKey.fromGroup(
+          new Group({ x: publicKeyFields[0], y: publicKeyFields[1] })
+        );
+        const messageFields = value.slice(2);
+        const messageChars = messageFields.map((f) => new Character(f));
+        const messageCS = CircuitString.fromCharacters(messageChars);
+        const message = messageCS.toString();
+        input = (
+          <>
+            <input key={i} onChange={onChange} />
+            <p>{`Offchain Value: ${message}`}</p>
+          </>
+        );
+        const signerLink =
+          'https://berkeley.minaexplorer.com/wallet/' + publicKey.toBase58();
+        signer = (
+          <a href={signerLink} target="_blank" rel="noreferrer">
+            <p>See signer</p>
+          </a>
+        );
+      } else {
+        input = <input key={i} onChange={onChange}></input>;
+      }
+      let button = <button onClick={makeOnUpdate(i)}> set </button>;
+      messages.push(
+        <div key={i}>
+          {' '}
+          {i}: {input} {signer} {button}
+        </div>
+      );
+    }
+    mainContent = (
+      <div>
+        <button onClick={onRefreshCurrentNum}> Get Latest State </button>
+        {messages}
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      {setup}
+      {accountDoesNotExist}
+      {mainContent}
+    </div>
```

Comment:
Suggestion:  If you wrap this with the GradientBG component included in the UI scaffold, you can achieve a similar UI theme to the tutorial 4 and the CLI UI landing pages.

If you want to do this. 

-  Import the `Home.module.css` styles in this file(copy the same styles from the file in tutorial 4)
-  Copy the `components` directory from tutorial 4(or the ui scaffold) as a sibling of the pages directory.
- Import the GradientBG component and wrap the UI in it like below.

```
    <GradientBG>
      <div className={styles.main} style={{ padding: 0 }}>
        <div className={styles.center} style={{ padding: 0 }}>
          {setup}
          {accountDoesNotExist}
          {mainContent}
        </div>
      </div>
    </GradientBG>
```

Reply:
<img width="1748" alt="Screen Shot 2023-08-30 at 2 11 12 AM" src="https://github.com/o1-labs/docs2/assets/2808242/27011ada-96cc-4d6b-9a78-509c4bbbd69e">

This is what it looks like